### PR TITLE
Implement custom Matcher for AssemblyOperator states

### PR DESCRIPTION
**Pull request by klalafaryan**
Signed-off-by: klalafaryan <lalafaryan@gmail.com>

### Type of change

- Enhancement / new feature

### Description

Implement custom Matcher for AssemblyOperator states.  Issue #3018 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by scholzj** [comment]
@klalafaryan Looks like there are some checkstyle issues. Can you have a look at them? 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:check (validate) on project cluster-operator: Failed during checkstyle execution: There are 2 errors reported by Checkstyle 8.29 with .checkstyle/checkstyle.xml ruleset. -> [Help 1]
```

**Review by samuel-hawker — APPROVED** [review]
LGTM!

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
  */
 package io.strimzi.operator.cluster.operator.assembly;
 
+import com.google.common.collect.Lists;
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java:7]: I don't think we want to add Guava as dependency. Can you solve this with regular Java lists/arrays? From the code it is not clear why is it really needed. This is also what is breaking the CI builds.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class AbstractResourceStateMatchers {
+
+
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java:22]: Do we need so many empty lines?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            @Override
+            public void describeTo(final Description description) {
+                //
+            }
+        };
+
+    }
+
+
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java:56]: Do we need so many empty lines?

**Review by scholzj — COMMENTED** [review]
I left some nits. But it looks good apart from them.

**Review by scholzj — APPROVED** [review]
Thanks for the PR.

**Pull request merged**